src/pages/newPost/NewPost.jsx

```diff
@@ -0,0 +1,128 @@
+import {useForm} from 'react-hook-form';
+import calculateReadTime from "../../helpers/calculateReadTime.jsx";
+import NewPostForm from "../../components/NewPostForm/NewPostForm.jsx";
+import "./NewPost.css";
+import {useState} from "react";
+import axios from "axios";
+import {Link} from "react-router-dom";
+import {useNavigate} from "react-router-dom";
+
+function NewPost() {
+    const {register, formState: {errors}} = useForm();
+    const [addPost, setAddPost] = useState([]);
```

Comment:
Als dit uiteindelijk een object gaat zijn, en dus geen array, dan zou ik een default waarde van `undefined` pakken, dan wordt je check verderop een stuk simpeler
```suggestion
    const [addPost, setAddPost] = useState();
```